processing-service-core/api/models/tenant.js

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright 2015-2017 EntIT Software LLC, a Micro Focus company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var Q = require('q');
+var logger = require('../helpers/loggingHelper.js');
+var tenantConfigStoreModel = require("./db/tenantConfigStore.js");
+var globalConfigStoreModel = require("./db/globalConfigStore.js");
+var apiErrorTypes = require('./errors/apiErrorTypes.js');
+var ApiError = require('./errors/apiError.js');
+
+
+module.exports = {
+    setTenantConfig: setTenantConfig,
+    setTenantConfigs: setTenantConfigs,
+    deleteTenantConfig: deleteTenantConfig,
+    deleteTenantConfigs: deleteTenantConfigs,
+    getEffectiveTenantConfig: getEffectiveTenantConfig,
+    getEffectiveTenantConfigs: getEffectiveTenantConfigs,
+    getTenantConfig: getTenantConfig,
+    getTenantConfigs: getTenantConfigs
+};
+
+/**
+ * Creates or updates tenant specific config.
+ * Creates or updates a custom configuration setting for this key against this tenant if one doesn't exist or updates the currently 
+ * existing custom configuration setting if it does.
+ *
+ * @param {String} tenantId String The ID of the tenant.
+ * @param {String} key String The custom configuration key.
+ * @param {String} value String The value to use when setting the custom config for the tenants key. (optional)
+ * no response value expected for this operation
+ **/
+function setTenantConfig(tenantId, key, value) {
+    var tenantConfig = Q.defer();
+    logger.debug("Creating tenant specific config for tenant " + tenantId + " with key " + key + " and value " + value);
+    tenantConfigStoreModel.setTenantConfig(tenantId, key, value)
+            .then(function (createConfigPromise) {
+                logger.debug("Configuration of " + key + " set for tenant " + tenantId + " with value " + value);
+                tenantConfig.resolve();
```

Comment:
We need a way to indicate if an item was created or deleted so the level above can determine the status code to set.

Reply:
Withdrawing that comment. Decision is to always return 204.